// react-socket-context/lib/SocketContext.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _socket = require('socket.io-client');

var _socket2 = _interopRequireDefault(_socket);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SocketContext = function (_Component) {
  _inherits(SocketContext, _Component);

  function SocketContext() {
    _classCallCheck(this, SocketContext);

    return _possibleConstructorReturn(this, (SocketContext.__proto__ || Object.getPrototypeOf(SocketContext)).apply(this, arguments));
  }

  _createClass(SocketContext, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      this.socket = (0, _socket2.default)(this.props.namespace);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this.socket.close();
      delete this.socket;
    }
  }, {
    key: 'getChildContext',
    value: function getChildContext() {
      return { socket: this.socket };
    }
  }, {
    key: 'render',
    value: function render() {
      return this.props.children;
    }
  }]);

  return SocketContext;
}(_react.Component);

SocketContext.childContextTypes = {
  socket: _react.PropTypes.object
};

SocketContext.propTypes = {
  namespace: _react.PropTypes.string,
  children: _react.PropTypes.element.isRequired
};

exports.default = SocketContext;
